src/GafferImage/OpenImageIOReader.cpp

```diff
@@ -1187,7 +1199,14 @@ void OpenImageIOReader::hash( const ValuePlug *output, const Context *context, I
 {
 	ImageNode::hash( output, context, h );
 
-	if( output == availableFramesPlug() )
+	if ( output == fileValidPlug() )
+	{
+		fileNamePlug()->hash( h );
+		refreshCountPlug()->hash( h );
+		channelInterpretationPlug()->hash( h );
+		hashFileName( context, h );
```

Comment:
I've added the context to the hash in order to get the frame change added, this might be too agressive though, thoughts?

Reply:
I think we do need to call `hashFileName()` to account for the frame number, but because we're doing that we shouldn't need to call `fileNamePlug()->hash( h )` as well.

Reply:
cleaned up and also added in missingFrameMode's hash as it now affects the fileValid plug